### PR TITLE
New term entry for Java Math.toIntExact()

### LINKED ISSUE #920: [Term Entry] Java Math Methods: `.toIntExact()`
### Reason/inspiration (optional)

Under `java/concepts/math-methods/terms/`, create a new term entry for the `Math.toIntExact()` Java method.

- In the `terms/` folder, make a new folder called `to-int-exact`
- Inside the `to-int-exact` folder, create a new markdown file called `to-int-exact.md`.
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Also make sure to include an "Example" section with a \```java code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [X] Issue is assigned

### PR DESCRIPTION
**Pull request by shivanshugour45**
### Description

This file present in "docs\content\java\concepts\math-methods\to-int-exact" contains the description of 'toIntExact()' method present in java.lang.Math library.

Closes #920  

### Type of Change

- Adding a new entry

### Checklist

- [v] All writings are my own.
- [v] My entry follows the Codecademy Docs style guide.
- [v] My changes generate no new warnings.
- [v] I have performed a self-review of my own writing and code.
- [v] I have checked my entry and corrected any misspellings.
- [v] I have made corresponding changes to the documentation if needed.
- [v] I have confirmed my changes are not being pushed from my forked `main` branch.
- [v] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [v] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The **`Math.toIntExact()`** method returns the integer value of a long type argument. Exception is thrown if the value of argument overflows that of an int.
+
+## Syntax
+
+```pseudo
+Math.toIntExact(long val)
+```
+
+The return type of the `.toIntExact()` method is an int type value. The data type of the `val` parameter is long.
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:24]:
```suggestion
The return type of the `.toIntExact()` method is an `int` type value. The data type of the `val` parameter is `long`.
```

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+- If the value of 'val' is greater than 2147483647L or is lesser than -2147483648L then an exception of following manner will be thrown.
+
+Exception in thread "main" java.lang.ArithmeticException: integer overflow
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:28]: Put the exception inside a ```shell block.

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+at java.base/java.lang.Math.toIntExact
+
+## Example
+
+The following example showcases the application of `.toIntExact()` method:
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:33]:
```suggestion
The following example demonstrates the application of `.toIntExact()` method:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@shivanshugour45, I did a first review and left some comments.

**Comment by shivanshugour45** [comment]
Hi @SSwiniarski, I have made the suggested changes.

**Review by shivanshugour45 — COMMENTED** [review]
All the suggested changes have been done.

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Syntax
+
+```pseudo
+Math.toIntExact(long val)
+```
+
+The return type of the `.toIntExact()` method is an `int` type value. The data type of the `val` parameter is `long`.
+
+- If the value of `val` is greater than 2147483647L or is lesser than -2147483648L then an exception of following manner will be thrown.
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:26]:
```suggestion
If the value of `val` is greater than 2147483647L or is lesser than -2147483648L then an exception of following manner will be thrown:
```

**Review by SSwiniarski — APPROVED** [review]
@shivanshugour45 I had one more comment, but it's otherwise ready for a second review.

**Review by SSwiniarski — APPROVED** [review]
@shivanshugour45, looks ready for review two.

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+The **`Math.toIntExact()`** method returns the integer value of a `long` type argument. An [exception](https://www.codecademy.com/resources/docs/java/errors) is thrown if the value of argument overflows that of an `int`.
+
+## Syntax
+
+```pseudo
+Math.toIntExact(long val)
+```
+
+The return type of the `.toIntExact()` method is an `int` type value. The data type of the `val` parameter is `long`.
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:24]: ~The return type of the `.toIntExact()` method is an `int` type value. The data type of the `val` parameter is `long`.~

Consider this instead:

"In the function `Math.toIntExact(long val)`, the parameter `val` is of type `long`. The function will return an `int`."

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+If the value of `val` is greater than 2147483647L or is lesser than -2147483648L then an exception of following manner will be thrown:
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:26]: "...or is ~lesser~ less than -2147483648L, then ~an exception of the following manner~ the following exception will be thrown."

**Review by HishamT — CHANGES_REQUESTED** [review]
@shivanshugour45, I have finished your second review and left some comments.

**Comment by shivanshugour45** [comment]
@HishamT, the requested changes are done.

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  - 'Arithmetic'
+  - 'Functions'
+  - 'Methods'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Math.toIntExact()`** method returns the integer value of a `long` type argument. An [exception](https://www.codecademy.com/resources/docs/java/errors) is thrown if the value of argument overflows that of an `int`.
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:16]: Remove word "argument".

**Review by HishamT — CHANGES_REQUESTED** [review]
@shivanshugour45, looks good! Just one minor change is requested.

**Comment by shivanshugour45** [comment]
@HishamT, the requested change is done.

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  public static void main(String args[]) {
+       long val= 2147483647L;
+       System.out.println(Math.toIntExact(val));
+   }
+}
+```
+
+This results in the following output:
+
+```java
```
Comment (yangc95) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:48]:
```suggestion
```shell
```

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+The **`Math.toIntExact()`** method returns the integer value of a `long` type. An [exception](https://www.codecademy.com/resources/docs/java/errors) is thrown if the argument's value overflows that of an `int`.
+
+## Syntax
+
+```pseudo
+Math.toIntExact(long val)
+```
+
+In the function `Math.toIntExact(long val)`, the parameter `val` is of type `long`. The function will return an `int`.
```
Comment (yangc95) [inline content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:24]: Can we use `value` instead of `val` here?

**Comment by yangc95** [comment]
@shivanshugour45 I also had some additional comments prior to merge

File: content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md (hunk 9 above)
Comment (shivanshugour45) [reply content/java/concepts/math-methods/terms/to-int-exact/to-int-exact.md:24]: Sure, will update the parameters now.

**Comment by shivanshugour45** [comment]
@yangc95, did the suggested changes. Please suggest further if any.

**Review by HishamT — APPROVED** [review]
Approved!

**Comment by yangc95** [comment]
@shivanshugour45 live here: https://www.codecademy.com/resources/docs/java/math-methods/toIntExact
Screen Shot 2022-10-22 at 3 11 06 PM
You're currently anonymous but if you ever wish to make it public, please add your GitHub to your Codecademy Profile.

cc: @HishamT @SSwiniarski

**Pull request merged**